app/controllers/openid_application_controller.rb

```diff
@@ -50,13 +50,13 @@ def authenticate_token
     if token.client_credentials_token? || token.ssoi_token?
       populate_payload_for_launch_patient_scope if token.payload['scp'].include?('launch/patient')
       populate_payload_for_launch_scope if token.payload['scp'].include?('launch')
-
       return true
     end
 
     return false if @session.uuid.nil?
 
     @current_user = OpenidUser.find(@session.uuid)
```

Comment:
Does this need to explicitly return false since it's no longer the last line in the method?

Reply:
My thought was the `confirm_icn_match` should catch that if nil and return false as well.

However, I did just add a `&.` to ensure the check is nil safe.